linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/FunctionEmitter.scala

```diff
@@ -518,8 +673,9 @@ private class FunctionEmitter private (
            * point, so we can trap as NPE.
            */
           markPosition(tree)
-          fb += wa.Unreachable
+          genNPE()
         } else {
+          genAsNonNullOrNPEFor(qualifier)
```

Comment:
Why not a check here? IIUC we do not need the as non null in unchecked mode? (it's implicit below).

Reply:
Indeed, a check is enough here.